larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/Cheating/ShowerDirectionCheater_tool.cc

```diff
@@ -120,15 +116,25 @@ namespace ShowerRecoTools {
         return 1;
       }
       trueParticle = trueParticles[ShowerTrackInfo.first];
+      ShowerEleHolder.SetElement(trueParticle,fTrueParticleInputLabel);
+    }
+
+    if (!trueParticle){
+      mf::LogError("ShowerDirectionCheater") << "True shower not found, returning";
+      return 1;
     }
 
-    TVector3 trueDir = {trueParticle->Px(),trueParticle->Py(),trueParticle->Pz()};
-    trueDir = trueDir.Unit(); // TODO: Can probably remove?
+    TVector3 trueDir = TVector3{trueParticle->Px(),trueParticle->Py(),trueParticle->Pz()}.Unit();
 
     TVector3 trueDirErr = {-999,-999,-999};
     ShowerEleHolder.SetElement(trueDir,trueDirErr,fShowerDirectionOutputLabel);
 
     if (fRMSFlip || fVertexFlip){
+
+      // Reset the tree values to defaults
+      rmsGradient = -5.f;
```

Comment:
Any particular motivation for these default values? (Default values are always a thorny issue, and not intending to look too deeply in this review. Can use +/- float max, or, even better, keep track of whether variables are explicitly assigned to, etc.)